subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/transform/DomainObjectContextProjectStateHandler.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.artifacts.transform;
+
+import org.gradle.api.Project;
+import org.gradle.api.internal.DomainObjectContext;
+import org.gradle.api.internal.artifacts.dsl.dependencies.ProjectFinder;
+import org.gradle.api.internal.project.ProjectState;
+import org.gradle.api.internal.project.ProjectStateRegistry;
+
+import javax.annotation.Nullable;
+
+public class DomainObjectContextProjectStateHandler {
```

Comment:
Does this handle the project state for the domain object _context,_ or for the domain object itself?

Reply:
I renamed it to `DomainObjectProjectStateHandler`.